### PR TITLE
remove client label from apiserver request count metric since it is unbounded

### PR DESCRIPTION
**Pull request by logicalhan**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

The client label is unbounded for the apiserver_request_total metric which can be problematic (memory-wise) as the process ages. Since the apiserver_request_total is a new metric and since it is an alpha metric, we are just going to shed this dimension

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Removed the 'client' label from apiserver_request_total.
```

/sig instrumentation
/assign @brancz

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 1]
```diff
@@ -65,14 +64,14 @@ var (
...
-			Help:           "Counter of apiserver requests broken out for each verb, dry run value, group, version, resource, scope, component, client, and HTTP response contentType and code.",
+			Help:           "Counter of apiserver requests broken out for each verb, dry run value, group, version, resource, scope, component, and HTTP response contentType and code.",
 			StabilityLevel: compbasemetrics.ALPHA,
 		},
 		// The label_name contentType doesn't follow the label_name convention defined here:
 		// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/instrumentation.md
 		// But changing it would break backwards compatibility. Future label_names
 		// should be all lowercase and separated by underscores.
-		[]string{"verb", "dry_run", "group", "version", "resource", "subresource", "scope", "component", "client", "contentType", "code"},
+		[]string{"verb", "dry_run", "group", "version", "resource", "subresource", "scope", "component", "contentType", "code"},
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: @mm4tt - can you please validate if it won't break anything in scalability tests?

**Review by wojtek-t — COMMENTED** [review]
I'm holding it to ensure that it won't break anything in scalability tests and our monitoring there...

/hold

@mm4tt @mborsz @oxddr @jkaniuk - FYI

**Comment by logicalhan** [comment]
For ref: https://github.com/kubernetes/kubernetes/issues/76302

**Comment by brancz** [comment]
Looks good from sig instrumentation perspective.

/lgtm

**Comment by mm4tt** [comment]
It'll cause inconveniences but won't break anything in our scale tests. We'll have to update a few dashboards but we can do it after this PR merges as they are only used for post-run analysis. 

To confirm, we won't be able to get the client anymore from the metric but we'll still be able to get it from kube-apiserver http logs, right?

/lgtm
/hold cancel

**Comment by wojtek-t** [comment]
> To confirm, we won't be able to get the client anymore from the metric but we'll still be able to get it from kube-apiserver http logs, right?

Yes.

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 2]
```diff
@@ -376,19 +374,6 @@ func cleanDryRun(u *url.URL) string {
 	return strings.Join(utilsets.NewString(dryRun...).List(), ",")
 }
 
-func cleanUserAgent(ua string) string {
```
Comment (lavalamp) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Consider instead making this choose between e.g. {"Browser", "kubectl", "self"}?

I'm happy to approve it either way but this avoids a change, has known cardinality, and remains at least a little useful.

Comment (logicalhan) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: After our offline discussion, do you mind if I mark this as resolved?

**Comment by lavalamp** [comment]
/approve

Seems like this label is pretty useless and we want to get rid of it anyway. SGTM

**Comment by Purushotham233** [comment]
@wojtek-t @logicalhan 
We were hoping for the client label of `apiserver_request_total` metric while analysing an outage. 

> To confirm, we won't be able to get the client anymore from the metric but we'll still be able to get it from kube-apiserver http logs, right?
> 
> Yes.

We have apiserver audit enabled but  only for  `create, update, delete` and for action on selected entities by human actor to limit the logging overhead to apisever and the cost of logs processing & storage. So not sure apiserver logs are really a practical alternative for this label. 

Please advise.

**Comment by yuzs2** [comment]
@logicalhan @lavalamp Do we have any alternative way to track the `client` of the apiserver request now?
We relied on the `client` label to collect the requests from different endpoints, since the `client` label has been removed, we cannot do that anymore.
Is there any alternative way or workaround? Please advise.

**Comment by lavalamp** [comment]
You can scan audit logs (or regular logs if you use --v >= 3).

**Pull request merged**